azurerm/internal/services/applicationinsights/application_insights_smart_detection_rule_resource.go

```diff
@@ -0,0 +1,186 @@
+package applicationinsights
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/applicationinsights/parse"
+
+	"github.com/Azure/azure-sdk-for-go/services/appinsights/mgmt/2015-05-01/insights"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceApplicationInsightsSmartDetectionRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApplicationInsightsSmartDetectionRuleUpdate,
+		Read:   resourceApplicationInsightsSmartDetectionRuleRead,
+		Update: resourceApplicationInsightsSmartDetectionRuleUpdate,
+		Delete: resourceApplicationInsightsSmartDetectionRuleDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Slow page load time",
+					"Slow server response time",
+					"Long dependency duration",
+				}, false),
+				DiffSuppressFunc: smartDetectionRuleNameDiff,
+			},
+
+			"application_insights_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"send_emails_to_subscription_owners": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+
+			"additional_emails": {
```

Comment:
would this make more sense as
```suggestion
			"addtional_emails_recipients": {
```

Reply:
Sure does!